### ISSUE TITLE
Latest V15 "Cannot find module koa"

### ISSUE THREAD

**Issue by TurtIeSocks** [jaydenseric/graphql-upload#314]
I'm upgrading from V13, followed the guide that demonstrated how to update the import paths. (I would recommend linking the GH issue in the patch notes maybe?) 

Info:
- TypeScript
- Express
- Apollo Server

Using the following:
```js
import graphqlUploadExpress from 'graphql-upload/graphqlUploadExpress'
```
```js
import GraphQLUpload from 'graphql-upload/GraphQLUpload'
```
```js
import type { FileUpload } from 'graphql-upload/processRequest'
```

Receiving the error:
```
../../../node_modules/graphql-upload/graphqlUploadKoa.js(45,21): error TS2307: Cannot find module 'koa' or its corresponding type declarations.
```
I see the following in `graphqlUploadKoa.js`, if it helps:
<img width="505" alt="Screen Shot 2022-05-28 at 11 46 47 AM" src="https://user-images.githubusercontent.com/58572875/170832649-15e37c6e-766d-4ef7-8cd2-209ed50e4ca0.png">

Any recommendations?

Thanks for the package!

**Comment by RemyMachado** [comment]
Same here:

`import { GraphQLUpload } from 'graphql-upload'`
> error TS2307: Cannot find module 'graphql-upload' or its corresponding type declarations.

**Comment by TurtIeSocks** [comment]
@RemyMachado if you change your import statement to what I have in the OP and remove the external `@types/graphql-upload` package, you should be able to get past that.

**Comment by RemyMachado** [comment]
@TurtIeSocks Unfortunately, no.

```typescript
import GraphQLUpload from 'graphql-upload/GraphQLUpload'
```

gives me:
`Could not find a declaration file for module 'graphql-upload/GraphQLUpload'`

I'm so annoyed by this library's import style.
I'm done. I removed the error with `// @ts-ignore`

**Comment by jaydenseric** [comment]
@TurtIeSocks thanks for the thanks about this package, it's appreciated.

Look in `node_modules/graphql-upload/`: There is no file called `GraphQLUpload`. So importing it can not be expected to magically work. The file is called [`GraphQLUpload.js`](https://github.com/jaydenseric/graphql-upload/blob/v15.0.0/GraphQLUpload.js).

You need to do this (with the `.js` in the path):

```js
import GraphQLUpload from "graphql-upload/GraphQLUpload.js";
```

The readme documents what modules are publicly exported, what format they are in, what there file names are, and links to the `package.json` file so you can see the package exports yourself:

<img width="791" alt="Screen Shot 2022-05-29 at 8 12 11 am" src="https://user-images.githubusercontent.com/1754873/170844376-e9ede383-9c1f-4a87-bc9f-274b8b2990b1.png">

https://github.com/jaydenseric/graphql-upload/blob/cd9e4f1c5d7cdf66173973a3637192b11c30796c/package.json#L39-L46

You can see more details about importing in this issue and comment:

https://github.com/jaydenseric/graphql-upload/issues/305#issuecomment-1140352014

Hopefully with the correct imports the TypeScript error regarding `graphqlUploadKoa.js` will go away. There is a chance though it's a seperate issue to resolve, and there are a few things to consider in that case.

**Comment by KillerCodeMonkey** [comment]
hey there. got the same problem. i think it is because you are linking all type defs in the preocessRequest.js

```
/** @typedef {import("./GraphQLUpload.js")} GraphQLUpload */
/** @typedef {import("./graphqlUploadExpress.js")} graphqlUploadExpress */
/** @typedef {import("./graphqlUploadKoa.js")} graphqlUploadKoa */
```

but if you are not using koa... or vice versa express you will get the error.

**Comment by KillerCodeMonkey** [comment]
so you need to add @types/koa to your dependencies as well.

**Comment by jaydenseric** [comment]
Thanks for finding the problem! Installing `koa` might not help as much as installing `@types/koa`. I'll remove that type import from `processRequest` and hopefully that will fix the issue.

The intention was that you should not need the Koa types installed if you are using Express, and vica versa. That's why those types are optional peer dependencies:

https://github.com/jaydenseric/graphql-upload/blob/cd9e4f1c5d7cdf66173973a3637192b11c30796c/package.json#L55-L62

**Comment by KillerCodeMonkey** [comment]
PS: you are write typings are enough. i do not know how exactly the typings with js works, so i thought it really imports source to extract the typings from the package. But nice that there is no overhead recarding in the production build size because only typings are needed during build time. so i get this going for me :D

**Comment by jaydenseric** [comment]
The fix has been published in [v15.0.1](https://github.com/jaydenseric/graphql-upload/releases/tag/v15.0.1) 🚀

**Comment by TurtIeSocks** [comment]
Confirmed, thanks @jaydenseric!
